proj1/generator.cpp

```diff
@@ -49,44 +50,39 @@ int main(int argc, char **argv) {
     argsParsing::checkAccess(argv[optind + 3], fstream::ios_base::out,
                              outputFile);
 
-    Model m(result.k, result.alpha, &trainFile);
+    Model m(result.k, result.alpha);
 
-    trainFile.close();
+    m.parseFile(trainFile);
 
-    map<string, map<char, double>> probs = m.getProbs();
+    trainFile.close();
 
-    // for (auto &i : probs) {
-    //     cout << i.first << endl;
-    //     for (auto &f : i.second) {
-    //         cout << f.first << ":" << f.second << ",";
-    //     }
-    //     cout << endl;
-    // }
+    auto probs = m.getStatsTable();
 
     bool wrote = false;
     char newChar;
     string context = initCtx;
     outputFile << context;
-    srand(time(0));
+
+    default_random_engine eng((random_device())());
+    uniform_real_distribution<double> prob_rand(0, 1);
+    uniform_int_distribution<int> idx_rand(0, m.getABC().size() - 1);
+
     for (int i = 0; i < numChars; i++) {
-        for (auto &tokens : probs) {
-            if (!((string)tokens.first).compare(context)) {
```

Comment:
If I understood correctly you wanted to iterate over the contexts until you find one matching the current context.
You can just check if the map has the key and if yes access it